src/com/MPIDirectCommunication.cpp

```diff
@@ -138,56 +143,52 @@ void MPIDirectCommunication::allreduceSum(int itemToSend, int &itemToReceive, Ra
   MPI_Allreduce(&itemToSend, &itemToReceive, 1, MPI_INT, MPI_SUM, _commState->comm);
 }
 
-void MPIDirectCommunication::broadcast(const int *itemsToSend, int size)
+void MPIDirectCommunication::broadcast(precice::span<const int> itemsToSend)
 {
-  PRECICE_TRACE(size);
-  MPI_Bcast(const_cast<int *>(itemsToSend), size, MPI_INT, 0, _commState->comm);
+  PRECICE_TRACE(itemsToSend.size());
+  MPI_Bcast(const_cast<int *>(itemsToSend.data()), itemsToSend.size(), MPI_INT, 0, _commState->comm);
 }
 
-void MPIDirectCommunication::broadcast(int *itemsToReceive,
-                                       int  size,
-                                       int  rankBroadcaster)
+void MPIDirectCommunication::broadcast(precice::span<int> itemsToReceive, int rankBroadcaster)
 {
-  PRECICE_TRACE(size);
-  MPI_Bcast(itemsToReceive, size, MPI_INT, rankBroadcaster, _commState->comm);
+  PRECICE_TRACE(itemsToReceive.size());
+  MPI_Bcast(itemsToReceive.data(), itemsToReceive.size(), MPI_INT, rankBroadcaster, _commState->comm);
 }
 
 void MPIDirectCommunication::broadcast(int itemToSend)
 {
   PRECICE_TRACE();
-  broadcast(&itemToSend, 1);
+  broadcast(precice::span<const int>{&itemToSend, 1});
 }
 
 void MPIDirectCommunication::broadcast(int &itemToReceive, Rank rankBroadcaster)
 {
   PRECICE_TRACE();
-  broadcast(&itemToReceive, 1, rankBroadcaster);
+  broadcast(precice::span<int>{&itemToReceive, 1}, rankBroadcaster);
 }
 
-void MPIDirectCommunication::broadcast(const double *itemsToSend, int size)
+void MPIDirectCommunication::broadcast(precice::span<const double> itemsToSend)
 {
-  PRECICE_TRACE(size);
-  MPI_Bcast(const_cast<double *>(itemsToSend), size, MPI_DOUBLE, 0, _commState->comm);
+  PRECICE_TRACE(itemsToSend.size());
+  MPI_Bcast(const_cast<double *>(itemsToSend.data()), itemsToSend.size(), MPI_DOUBLE, 0, _commState->comm);
```

Comment:
Why `0`? Could you replace this magic number with something more descriptive?

Reply:
True, but a bit too off topic.